scipy/stats/_multivariate.py

```diff
@@ -5067,3 +5068,257 @@ def rvs(self, size=1, random_state=None):
+
+    Parameters
+    ----------%(_doc_callparams)s
+    %(_doc_random_state)s
+
+    Notes
+    -----
+    %(_doc_callparams_note)s
+
+    Random elements from the distribution are generated either with a shuffling
+    algorithm or with Patefield's algorithm [1]_. The shuffling algorithm has
+    O(N) time and space complexity, where N is the total sum of entries in the
+    table. Patefield's algorithm has O(K x log(N)) time complexity, where K is
+    the number of cells in the table and requires only a small constant work
+    space.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.stats import conditional_table
+
+    >>> row = [1, 5]
+    >>> col = [2, 3, 1]
+    >>> conditional_table.mean(row, col)
+    array([[0.33333333, 0.5       , 0.16666667],
+           [1.66666667, 2.5       , 0.83333333]])
+
+    Alternatively, the object may be called (as a function) to fix the row
+    and column vector sums, returning a "frozen" distribution.
+
+    >>> dist = conditional_table(row, col)
+    >>> dist.rvs(size=1, random_state=123)
+    array([[[1., 0., 0.],
+            [1., 3., 1.]]])
+
+    .. versionadded: 1.9.3
+
+    References
+    ----------
+    .. [1] Patefield's, AS 159 Appl. Statist. (1981) vol. 30, no. 1.
+    """
+
+    def __init__(self, seed=None):
+        super().__init__(seed)
+
+    def __call__(self, row, col, seed=None):
+        """Create a frozen distribution of tables with fixed column and row marginals.
+
+        See `conditional_table_frozen` for more information.
+        """
+        return conditional_table_frozen(row, col, seed)
+
+    def _process_parameters(self, row, col):
+        """
+        Check that row and column vectors are one-dimensional, that they do not
+        contain negative entries, and that the sums over both vectors are equal.
+        """
+        r = np.array(row, dtype=np.int_, copy=True)
+        c = np.array(col, dtype=np.int_, copy=True)
+
+        if np.ndim(r) != 1:
+            raise ValueError("row array must be one-dimensional")
+        if np.ndim(c) != 1:
+            raise ValueError("column array must be one-dimensional")
+
+        if np.any(r < 0):
+            raise ValueError("each element of row array must be non-negative")
+
+        if np.any(c < 0):
+            raise ValueError("each element of column array must be non-negative")
+
+        ntot_row = np.sum(row)
+        ntot_col = np.sum(col)
+        if ntot_row != ntot_col:
+            raise ValueError("Sums over all rows and columns must be equal.")
+
+        return r, c, ntot_row
+
+    def mean(self, row, col):
+        """Mean of distribution of conditional tables.
+        %(_doc_mean_callparams)s
+        Returns
+        -------
+        mean: ndarray
+            The mean of the distribution
+
+        Notes
+        -----
+        %(_doc_callparams_note)s
+        """
+        r, c, n = self._process_parameters(row, col)
+        return np.outer(r, c) / n
```

Comment:
I don't have a reference for this, but it has to be like this if you assume rows and columns to be independent. The expectation for each bin is the number of entries times the frequency per bin. The frequency per bin, since rows and columns are independent, must be the product of the row and column frequencies.

Reply:
Got it, thanks!